### PR TITLE
Added shortName to meta.v1.APIResource

### PR DESCRIPTION
**Pull request by p0lyn0mial**
**Special notes for your reviewer**:

I have followed the steps described here [Changing the internal structures](https://github.com/kubernetes/community/blob/master/contributors/devel/api_changes.md )

I have run the following scripts:

hack/update-codegen.sh
hack/update-generated-protobuf.sh
hack/update-codecgen.sh
hack/update-openapi-spec.sh
hack/update-swagger-spec.sh
hack/update-federation-openapi-spec.sh

**See also**:
1. https://github.com/kubernetes/kubernetes/pull/38755
2. https://github.com/kubernetes/kubernetes/pull/38835#event-914403188

### CODE REVIEW

**Comment by p0lyn0mial** [comment]
@deads2k can you please take a look at this ?

**Comment by deads2k** [comment]
@k8s-bot test this

File: staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go [hunk 1]
```diff
@@ -579,6 +579,8 @@ type APIResource struct {
 	// verbs is a list of supported kube verbs (this includes get, list, watch, create,
 	// update, patch, delete, deletecollection, and proxy)
 	Verbs Verbs `json:"verbs" protobuf:"bytes,4,opt,name=verbs"`
+	// shortName is the suggested short name of the resource.
+	ShortName string `json:"shortName, omitempty" protobuf:"bytes,5,opt,name=shortName"`
```
Comment (deads2k) [inline staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go]: @smarterclayton @lavalamp @liggitt this is the meat of the change right here.  This `shortName` would be used to allow `kubectl` to have short name representations of resources from aggregated API servers.  @p0lyn0mial already updated the shortcut expander to be group aware, so collisions won't be a problem.

**Comment by deads2k** [comment]
I like the change (thanks for keeping it small) and given your previous refactors, this ought to plug in nicely.

**Comment by deads2k** [comment]
Looks like two more update scripts: `hack/update-api-reference-docs.sh` and `hack/update-swagger-spec.sh`.  (clicked on the verification failure details, searched for "failed ")

**Comment by deads2k** [comment]
@k8s-bot unit test this: issue #39471

**Comment by deads2k** [comment]
gke is currently broken in head, so I wouldn't worry about those.

**Comment by p0lyn0mial** [comment]
@deads2k can you kick off the test - just want to make sure that all generators have been run and then I will do rebase.

**Comment by deads2k** [comment]
> @deads2k can you kick off the test - just want to make sure that all generators have been run and then I will do rebase.

Without a rebase, all tests will fail.  The test run by checking out master and then apply your patches.

**Comment by deads2k** [comment]
When you rebase, go ahead and squash.  I'll bug daniel and clayton today to get a second lgtm.

**Comment by p0lyn0mial** [comment]
so sorry for closing this PR - something has broken while rebasing and merging. It's easier for me to create a new branch (coming soon)

**Comment by deads2k** [comment]
@lavalamp he added this

```go
ShortName string `json:"shortName, omitempty" protobuf:"bytes,5,opt,name=shortName"
```

To `APIResource` here: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go#L572-L582

**Comment by lavalamp** [comment]
Oh, yes, we talked about something like this before. Was I arguing that it should be a list? Did you convince me otherwise?

**Comment by deads2k** [comment]
I was arguing it should be a list, you argued against.  I didn't feel strongly.  I said list so that an API author could transition his shortcut names.  You said that only mattered for scripts and people shouldn't use shortcuts there.

**Pull request closed without merging**